### PR TITLE
Add workflow for slow builds

### LINKED ISSUE #211: Periodically Run Expensive Tests
#202 / #208 test the image bootstrapping system. A subset of tests is all that should run on a per check in basis.

The suffix `expensive` was added to the test name (e.g. `test_cdt2_expensive`) as an indicator that maybe it shouldn't run on every commit. Additionally, the argument `-k not expensive` to the workflow pylint command.

@mikelik correctly makes the argument that disabling these tests in CI can allow a regression to slip by us that would otherwise be caught.

One option to resolve this is to run the tests periodically. Maybe once a week or even once a month.

### PR DESCRIPTION
**Pull request by ScottBailey**
1. Renamed existing workflow for short tests to be clearer.
2. Added a workflow to run all tests.

Will close #211.

### CODE REVIEW

**Review by mikelik — APPROVED** [review]
Will this run both `all-tests.yaml` and `run-tests.yaml` after merging to main or release?
Nice to have would be to run `run-tests.yaml` only if the merge is not to main/release, but I'm fine with current version.

**Comment by ScottBailey** [comment]
> Will this run both `all-tests.yaml` and `run-tests.yaml` after merging to main or release? Nice to have would be to run `run-tests.yaml` only if the merge is not to main/release, but I'm fine with current version.

@mikelik, I made some additional changes (please re-review! TIA) and I expect to it to run the short tests on all PRs and the long test on all pushes (including merge) to main. Additionally, long or short test can be run manually.

Unfortunately, I've only been able to partially test this in the branch. After merge we will need to review that everything works as expected.

File: .github/workflows/run-short-tests.yaml [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - name: Use Docker in rootless mode (To solve "Permission denied" error when removing test_app directory)
-        uses: ScribeMD/rootless-docker@0.2.2
-      - uses: actions/checkout@v2
+      # Use Docker in rootless mode (To solve "Permission denied" error when removing test_app directory)
+      - uses: ScribeMD/rootless-docker@0.2.2
```
Comment (mikelik) [inline .github/workflows/run-short-tests.yaml:15]:
```suggestion
      - name: Use Docker in rootless mode (To solve "Permission denied" error when removing test_app directory)
        uses: ScribeMD/rootless-docker@0.2.2
```
name is more useful, as it is printed in the log

File: .github/workflows/run-short-tests.yaml [hunk 2]
```diff
@@ -1,16 +1,21 @@
-name: Run tests
+name: Run Short Tests
 
-on: [push]
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+  workflow_dispatch:
```
Comment (mikelik) [inline .github/workflows/run-short-tests.yaml:8]:
```suggestion
on: [pull_request, workflow_dispatch]
```
I think we are fine with default types of pull_request (which are `opened`, `synchronize`, `reopened`)

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

**Review by mikelik — APPROVED** [review]

File: .github/workflows/run-short-tests.yaml (hunk 2 above)
Comment (ScottBailey) [reply .github/workflows/run-short-tests.yaml:8]: Pull request types include `edited`, `labeled`, `assigned`, `review_requested`, and others that would needlessly start a build. I think. So I prefer to leave it as is.

File: .github/workflows/run-short-tests.yaml (hunk 1 above)
Comment (ScottBailey) [reply .github/workflows/run-short-tests.yaml]: Taking suggestion.

File: .github/workflows/run-short-tests.yaml (hunk 2 above)
Comment (ScottBailey) [reply .github/workflows/run-short-tests.yaml:8]: I did not see that! Taking the suggestion!

**Review by mikelik — APPROVED** [review]

**Review by jolly-fellow — APPROVED** [review]

**Pull request merged**